src/extensions/default/CodeFolding/foldhelpers/foldgutter.js

```diff
@@ -0,0 +1,308 @@
+/**
+ * Based on http://codemirror.net/addon/fold/foldgutter.js
+ * @author Patrick Oladimeji
+ * @date 10/24/13 10:14:01 AM
+ */
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, brackets, document, window, $*/
+define(function (require, exports, module) {
+    "use strict";
+    var CodeMirror      = brackets.getModule("thirdparty/CodeMirror2/lib/codemirror"),
+        prefs           = require("Prefs");
+
+    function State(options) {
+        this.options = options;
+        this.from = this.to = 0;
+    }
+
+    function parseOptions(opts) {
+        if (opts === true) { opts = {}; }
+        if (!opts.gutter) { opts.gutter = "CodeMirror-foldgutter"; }
+        if (!opts.indicatorOpen) { opts.indicatorOpen = "CodeMirror-foldgutter-open"; }
+        if (!opts.indicatorFolded) { opts.indicatorFolded = "CodeMirror-foldgutter-folded"; }
+        return opts;
+    }
+
+    /**
+        Utility for creating fold markers in fold gutter
+        @param {string} spec the className for the marker
+        @returns {HTMLElement}
+    */
+    function marker(spec) {
+        var elt = document.createElement("div");
+        elt.className = spec;
+        return elt;
+    }
+
+    /**
+     Updates the gutter markers for the specified range
+     @param {!CodeMirror} cm the CodeMirror instance for the active editor
+     @param {!number} from the starting line for the update
+     @param {!number} to the ending line for the update
+    */
+    function updateFoldInfo(cm, from, to) {
+        var minFoldSize = prefs.getSetting("minFoldSize") || 2;
+        var opts = cm.state.foldGutter.options;
+        var fade = prefs.getSetting("fadeFoldButtons");
+        var gutter = $(cm.getGutterElement());
+        var i = from;
+        var isFold = function (m) {
+            return m.__isFold;
+        }, clear = function (m) {return m.clear(); };
+
+        /**
+            helper function to check if the given line is in a folded region in the editor.
+            @param {Number} line the
+            @return {from: {ch: Number, line: Number}, to: {ch: Number, line: Number}} the range that hides the specified line or undefine if the line is not hidden
+        */
+        function _isCurrentlyFolded(line) {
+            var keys = Object.keys(cm._lineFolds), i = 0, r;
```

Comment:
Can we please name `r` to a little more descriptive name?  It helps us mere humans read the code a bit better :)